pymc_marketing/mmm/risk_assessment.py

```diff
@@ -0,0 +1,461 @@
+
+    The Conditional Value at Risk (CVaR) is calculated as:
+
+        .. math::
+            CVaR = \mathbb{E}[X \mid X \leq VaR]
+
+    where :math:`X` represents the loss distribution, and :math:`VaR` is the Value at Risk
+    at the specified confidence level. CVaR provides a more comprehensive view of the risk
+    associated with extreme losses beyond the VaR.
+
+    Parameters
+    ----------
+    confidence_level : float, optional
+        Confidence level for CVaR (default is 0.95).
+
+    Returns
+    -------
+    ObjectiveFunction
+        A function that calculates the CVaR value at the specified confidence level given samples and budgets.
+
+    Raises
+    ------
+    ValueError
+        If confidence_level is not between 0 and 1.
+        If no samples fall below the VaR threshold.
+
+    References
+    ----------
+    - Rockafellar, R.T., & Uryasev, S. (2000). Optimization of Conditional Value-at-Risk.
+    """
+
+    def _conditional_value_at_risk(samples: np.ndarray, budgets: np.ndarray) -> float:
+        if not 0 < confidence_level < 1:
+            raise ValueError("Confidence level must be between 0 and 1.")
+        VaR = np.percentile(samples, (1 - confidence_level) * 100)
+        tail_losses = samples[samples <= VaR]
+        if len(tail_losses) == 0:
+            raise ValueError(
+                "No samples fall below the VaR threshold; CVaR is undefined."
+            )
+        CVaR = tail_losses.mean()
+        return CVaR
+
+    return _conditional_value_at_risk
+
+
+def sharpe_ratio(risk_free_rate: float = 0.0) -> ObjectiveFunction:
+    R"""
+    Calculate the Sharpe Ratio.
+
+    The Sharpe Ratio assesses the risk-adjusted return of an investment by comparing
+    the excess return over the risk-free rate to the standard deviation of returns.
+
+    The Sharpe Ratio is calculated as:
+
+        .. math::
+            Sharpe\ Ratio = \frac{\mathbb{E}[R - R_f]}{\sigma}
+
+    where:
+        - :math:`\mathbb{E}[R - R_f]` is the mean of excess returns.
+        - :math:`\sigma` is the standard deviation of the excess returns.
+
+    Parameters
+    ----------
+    risk_free_rate : float, optional
+        Risk-free rate of return (default is 0.0).
+
+    Returns
+    -------
+    ObjectiveFunction
+        A function that calculates the Sharpe Ratio given samples and budgets.
+
+    Raises
+    ------
+    ValueError
+        If the standard deviation of excess returns is zero.
+
+    References
+    ----------
+    - Sharpe, W.F. (1966). Mutual Fund Performance.
+    """
+
+    def _sharpe_ratio(samples: np.ndarray, budgets: np.ndarray) -> float:
+        excess_returns = samples - risk_free_rate
+        mean_excess_return = np.mean(excess_returns)
+        std_excess_return = np.std(excess_returns, ddof=1)
+        if std_excess_return == 0:
+            raise ValueError(
+                "Standard deviation of excess returns is zero; Sharpe Ratio is undefined."
+            )
```

Comment:
This is have an issue with floating point precision. Let's remove or check differently